src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -194,10 +194,9 @@ protected AIState decide()
             }
         }
 
-        if (worker.getCitizenColonyHandler().getWorkBuilding() != null
-              && !(worker.getLastAttackedEntity() != null
-                     && !worker.getLastAttackedEntity().isDead)
-              && getOwnBuilding() instanceof AbstractBuildingGuards
+        if (!(worker.getLastAttackedEntity() != null
+              && !worker.getLastAttackedEntity().isDead)
+              && getOwnBuilding(AbstractBuildingGuards.class) != null
               && target == null)
         {
             final AbstractBuildingGuards guardBuilding = (AbstractBuildingGuards) getOwnBuilding();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast to "AbstractBuildingGuards". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1905)